src/traffic_server/InkAPI.cc

```diff
@@ -9004,6 +9005,169 @@ TSSslContextDestroy(TSSslContext ctx)
   SSLReleaseContext(reinterpret_cast<SSL_CTX *>(ctx));
 }
 
+TSReturnCode
+TSSslClientCertUpdate(const char *cert_path, const char *key_path)
+{
+  if (nullptr == cert_path) {
+    return TS_ERROR;
+  }
+
+  std::string key;
+  SSL_CTX *test_ctx       = nullptr;
+  SSLConfigParams *params = SSLConfig::acquire();
+
+  // Generate second level key for client context lookup
+  ts::bwprint(key, "{}:{}", cert_path, key_path);
+  Debug("ssl.cert_update", "TSSslClientCertUpdate(): Use %.*s as key for lookup", static_cast<int>(key.size()), key.data());
+
+  // Use same file for key and cert if key file is not provided
+  if (nullptr == key_path) {
+    key_path = cert_path;
+  }
+
+  if (nullptr != params) {
+    // First try to update a default context with given cert/key to verify
+    test_ctx = SSL_CTX_new(SSLv23_client_method());
+    if (!SSL_CTX_use_certificate_chain_file(test_ctx, cert_path)) {
+      SSLError("failed to load client certificate from %s", cert_path);
+      SSL_CTX_free(test_ctx);
+      return TS_ERROR;
+    }
+    if (!SSL_CTX_use_PrivateKey_file(test_ctx, key_path, SSL_FILETYPE_PEM)) {
```

Comment:
Don't you have to back off and try the cert file if the private key fails?

Reply:
I think the current logic relies on caller to know what exactly it wants to do: if use same file for both, then it should pass in nullptr as `key_path` (which gets set to `cert_path` at line 9024.